File: nom-ui/src/app/recipe/models/recipe-search.model.ts
export interface RecipeSearchModel {
    query?: string;
    ingredientIds?: number[];
    categoryIds?: number[];
    tagIds?: number[];
    toolIds?: number[];
    cuisineTypeIds?: number[];
    minRating?: number;
    maxPrepTime?: number;
    maxCookTime?: number;
    maxTotalTime?: number;
    isPublic?: boolean;
    isApproved?: boolean;
    sortBy?: string;
    sortDirection?: string;
    page: number;
    pageSize: number;
    includeIngredients: boolean;
    includeSteps: boolean;
    includeNutrition: boolean;
}

export interface RecipeSearchResponse {
    recipes: RecipeSearchResult[];
    results: RecipeSearchResult[]; // Alias for recipes
    totalCount: number;
    page: number;
    pageSize: number;
    totalPages: number;
    hasNextPage: boolean;
    hasPreviousPage: boolean;
}

export interface RecipeSearchResult {
    id: number;
    name: string;
    description?: string;
    imageUrl?: string;
    prepTime: number;
    cookTime: number;
    totalTime: number;
    servings: number;
    averageRating: number;
    ratingCount: number;
    rating: number; // Alias for averageRating
    isPublic: boolean;
    isApproved: boolean;
    createdDate: Date;
    lastModifiedDate?: Date;
    authorName: string;
    categories: string[];
    tags: string[];
    cuisineTypes: string[];
    ingredients?: RecipeIngredientSearch[];
    steps?: RecipeStepSearch[];
    nutrition?: RecipeNutritionSearch;
}

export interface RecipeIngredientSearch {
    id: number;
    name: string;
    quantity?: number;
    measurement?: string;
    notes?: string;
}

export interface RecipeStepSearch {
    id: number;
    stepNumber: number;
    instructions: string;
    imageUrl?: string;
}

export interface RecipeNutritionSearch {
    calories?: number;
    protein?: number;
    carbohydrates?: number;
    fat?: number;
    fiber?: number;
    sugar?: number;
    sodium?: number;
} 

File: nom-api/Nom.Orch/Models/Recipe/RecipeSearchModel.cs
using System.ComponentModel.DataAnnotations;

namespace Nom.Orch.Models.Recipe
{
    public class RecipeSearchModel
    {
        [StringLength(255, ErrorMessage = "Search query cannot exceed 255 characters.")]
        public string? Query { get; set; }
        
        public List<long>? IngredientIds { get; set; }
        public List<long>? CategoryIds { get; set; }
        public List<long>? TagIds { get; set; }
        public List<long>? ToolIds { get; set; }
        public List<long>? CuisineTypeIds { get; set; }
        
        public int? MinRating { get; set; }
        public int? MaxPrepTime { get; set; }
        public int? MaxCookTime { get; set; }
        public int? MaxTotalTime { get; set; }
        
        public bool? IsPublic { get; set; }
        public bool? IsApproved { get; set; }
        
        public string? SortBy { get; set; } // "name", "rating", "date", "prepTime", "cookTime"
        public string? SortDirection { get; set; } // "asc", "desc"
        
        public int Page { get; set; } = 1;
        public int PageSize { get; set; } = 20;
        
        public bool IncludeIngredients { get; set; } = true;
        public bool IncludeSteps { get; set; } = false;
        public bool IncludeNutrition { get; set; } = false;
    }
} 